src/libcore/sync/atomic.rs

```diff
@@ -1418,6 +1419,128 @@ assert_eq!(foo.load(Ordering::SeqCst), 0b011110);
                     unsafe { atomic_xor(self.v.get(), val, order) }
                 }
             }
+
+            doc_comment! {
+                concat!("Fetches the value, and applies a function to it that returns an optional
+new value. Returns a `Result` (`Ok(_)` if the function returned `Some(_)`, else `Err(_)`) of the
+previous value.
+
+Note: This may call the function multiple times if the value has been changed from other threads in
+the meantime, as long as the function returns `Some(_)`. It will however only apply the function
+once to the value stored within.
```

Comment:
> It will however only apply the function once to the value stored within.

I don't get what does this mean. Could be clarify this in the code?

Reply:
I wanted to say that the function will be applied only once per observed value.

Reply:
That is true. Ok, I've changed the text once more to say that the function will only have been applied once to the *stored value*. Otherwise people may fear they need to make the function idempotent.